linux bridge: switch to nispor for querying

cathay4t opened the pull request:
Use nispor for bridge status query(including vlan filtering) and remove
the query code from NetworkManager plugin.

Since both nispor and NetworkManager plugin need to convert
nmstate state to kernel port VLAN format, the
`ifaces/linux_bridge_port_vlan.py` is added to hold the share functions.

The kernel will automatically compress adjacent VLAN IDs to VLAN range,
at the verification stage, `LinuxBridgeIface` will expand VLAN range to
single VLAN IDs.

Also move linux bridge only functions to `LinuxBridgeIface()`.

Unit test cases added for `ifaces/linux_bridge_port_vlan.py`.

cathay4t commented:
@maiqueb Could you review this PR?

The main purpose is querying base on kernel runtime status for bridge instead of reading NetworkManager on-disk config.

You may focus the review against `libnmstate/ifaces/linux_bridge_port_vlan.py` and `libnmstate/nm/bridge_port_vlan.py`.

Thank you!

maiqueb commented:
> @maiqueb Could you review this PR?
> 
> The main purpose is querying base on kernel runtime status for bridge instead of reading NetworkManager on-disk config.
> 
> You may focus the review against `libnmstate/ifaces/linux_bridge_port_vlan.py` and `libnmstate/nm/bridge_port_vlan.py`.
> 
> Thank you!

I'll reply in the next few days, but on the short look I've taken, it looks OK.

ffmancera commented on libnmstate/ifaces/linux_bridge_port_vlan.py:
```diff
@@ -0,0 +1,176 @@
+from libnmstate.schema import LinuxBridge as LB
+
+
+class KernelBridgePortVlans:
+    def __init__(
+        self,
+        vid_min=None,
+        vid_max=None,
+        is_pvid=False,
+        is_egress_untagged=False,
+    ):
+        self.vid_min = vid_min
+        self.vid_max = vid_max
+        self.is_pvid = is_pvid
+        self.is_egress_untagged = is_egress_untagged
+
+    def get_vlan_tag_range(self):
+        return (self.vid_min, self.vid_max)
+
+    def to_dict(self):
+        if self.vid_min != self.vid_max:
+            return {
+                LB.Port.Vlan.TrunkTags.ID_RANGE: {
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE: self.vid_min,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: self.vid_max,
+                }
+            }
+        else:
+            return {LB.Port.Vlan.TrunkTags.ID: self.vid_min}
```
I understand the need of this class but not the module. I would expect `libnmstate/ifaces/` contains one module for each interface, port vlan filtering is a linux_bridge feature, not an interface itself. What do you think?

cathay4t replied:
Yes. But that will make `ifaces/linux_bridge.py` holding 500+ lines. Just look better when splitting into two files.

ffmancera commented on libnmstate/nispor/bridge_port_vlan.py:
```diff
@@ -0,0 +1,37 @@
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+
+from libnmstate.ifaces import NmstateLinuxBridgePortVlan
+from libnmstate.ifaces import KernelBridgePortVlans
+
+
+def get_port_vlan_info(np_sub):
+    return NmstateLinuxBridgePortVlan.new_from_kernel_vlans(
+        [_np_vlan_to_kernel_vlan(np_vlan) for np_vlan in np_sub.vlans]
+    ).to_dict()
+
+
+def _np_vlan_to_kernel_vlan(np_vlan):
+    if np_vlan.vid_range:
+        vid_min, vid_max = np_vlan.vid_range
+    else:
+        vid_min = vid_max = np_vlan.vid
+    return KernelBridgePortVlans(
+        vid_min, vid_max, np_vlan.is_pvid, np_vlan.is_egress_untagged
```
In this case I am fine with putting this into its own file.

ffmancera commented on libnmstate/nispor/plugin.py:
```diff
@@ -66,7 +66,13 @@ def get_interfaces(self):
                 ifaces.append(NisporPluginVxlanIface(np_iface).to_dict())
             elif iface_type == "Bridge":
-                ifaces.append(NisporPluginBridgeIface(np_iface).to_dict())
+                np_ports = []
+                for port_name in np_iface.ports:
+                    if port_name in np_state.ifaces.keys():
+                        np_ports.append(np_state.ifaces[port_name])
```
Could this be done inside the init of `NisporPluginBridgeIface`? Thanks!

cathay4t replied:
If I do that, the `NisporPluginBridgeIface` need to access `np_state.ifaces` which it should not be.

cathay4t committed:
linux bridge: switch to nispor for querying

Use nispor for bridge status query(including vlan filtering) and remove
the query code from NetworkManager plugin.

Since both nispor and NetworkManager plugin need to convert
nmstate state to kernel port VLAN format, the
`ifaces/linux_bridge_port_vlan.py` is added to hold the share functions.

The kernel will automatically compress adjacent VLAN IDs to VLAN range,
at the verification stage, `LinuxBridgeIface` will expand VLAN range to
single VLAN IDs.

Also move linux bridge only functions to `LinuxBridgeIface()`.

Unit test cases added for `ifaces/linux_bridge_port_vlan.py`.

Signed-off-by: Gris Ge <fge@redhat.com>

maiqueb reviewed (approved):
I'm good w/ the proposed changes (focused exclusively on the vlan filter aspects).
